src/hooks/use-filtered-assets-list.hook.ts

```diff
@@ -1,16 +1,39 @@
+import { BigNumber } from 'bignumber.js';
 import { useEffect, useState } from 'react';
 
+import { useExchangeRatesSelector } from '../store/currency/currency-selectors';
 import { TokenInterface } from '../token/interfaces/token.interface';
+import { getTokenSlug } from '../token/utils/token.utils';
+import { isDefined } from '../utils/is-defined';
 import { isString } from '../utils/is-string';
-import { isNonZeroBalance } from '../utils/tezos.util';
+import { isNonZeroBalance, mutezToTz } from '../utils/tezos.util';
 
 export const useFilteredAssetsList = (assetsList: TokenInterface[], initialIsHideZeroBalance = false) => {
+  const exchangeRates = useExchangeRatesSelector();
+
   const [isHideZeroBalance, setIsHideZeroBalance] = useState(initialIsHideZeroBalance);
   const [nonZeroBalanceAssetsList, setNonZeroBalanceAssetsList] = useState<TokenInterface[]>([]);
 
   const [searchValue, setSearchValue] = useState<string>();
   const [filteredAssetsList, setFilteredAssetsList] = useState<TokenInterface[]>([]);
 
+  useEffect(() => {
+    assetsList.sort((asset1, asset2) => {
+      const zero = new BigNumber(0);
+
+      const exchangeRate1: number | undefined = exchangeRates[getTokenSlug(asset1)];
+      const exchangeRate2: number | undefined = exchangeRates[getTokenSlug(asset2)];
+
+      const parsedAmount1 = mutezToTz(new BigNumber(asset1.balance), asset1.decimals);
+      const parsedAmount2 = mutezToTz(new BigNumber(asset2.balance), asset2.decimals);
+
+      const dollarAmount1 = isDefined(exchangeRate1) ? parsedAmount1.multipliedBy(exchangeRate1) : zero;
+      const dollarAmount2 = isDefined(exchangeRate2) ? parsedAmount2.multipliedBy(exchangeRate2) : zero;
+
+      return dollarAmount2.minus(dollarAmount1).toNumber();
+    });
+  }, [assetsList]);
+
```

Comment:
1. Sorting assets by dollar price should be applied at the Home screen only, but this hook is used in other places too.
2. It's a bad practice to mutate the array that you are working with. Instead of this try to memoize sorted copy of array.